plugins/org.eclipse.fordiac.ide.export.forte_ng/src/org/eclipse/fordiac/ide/export/forte_ng/ForteFBTemplate.xtend

```diff
@@ -377,20 +379,35 @@ abstract class ForteFBTemplate<T extends FBType> extends ForteLibraryElementTemp
 
 	def protected generateSetInitialValuesDeclaration(Iterable<VarDeclaration> variables) '''
 		«IF !variables.empty»
-			void setInitialValues() override;
+			void setInitialValues(bool retain) override;
```

Comment:
Why do we need a new boolean parameter here? Is there a case where we do not want to retain variables marked as *RETAIN*?

Reply:
Yes we need the boolean argument. But it should be called cold. To indicate the right meaning of this variable.